src/org/nejrasm/zadaca3oop/task2/ElectricCar.java

```diff
@@ -0,0 +1,33 @@
+package org.nejrasm.zadaca3oop.task2;
+
+public class ElectricCar extends Vehicle {
+    private final ElectricEngine electricCarEngine;
+
+    public ElectricCar(final String model, final double capacity) {
+        super(model);
+        this.electricCarEngine = new ElectricEngine(capacity);
+    }
+
+    @Override
+    public double accelerate(final double acceleration) {
+        final double capacityState = this.electricCarEngine.decreaseCapacity(0.4 * acceleration / 100);
```

Comment:
The capacity factor should be part for the ElectricEngine class. 

The vehicle should not know details about it's engine, and the engine should handle fuel/capacity management.

I say this, because if there was an ElectricBike class reusing ElectricEngine we would need to provide this value again.

Reply:
Fixed. The attribute is private and it is adjustable by getters and setters.